script/split_tests.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python3
+"""Helper script to split test into n buckets."""
+
+from __future__ import annotations
+
+import argparse
+from dataclasses import dataclass, field
+from math import ceil
+from pathlib import Path
+import subprocess
+import sys
+from typing import Final
+
+
+class Bucket:
+    """Class to hold bucket."""
+
+    def __init__(
+        self,
+    ):
+        """Initialize bucket."""
+        self.total_tests = 0
+        self._paths: list[str] = []
+
+    def add(self, part: TestFolder | TestFile) -> None:
+        """Add tests to bucket."""
+        part.add_to_bucket()
+        self.total_tests += part.total_tests
+        self._paths.append(str(part.path))
+
+    def get_paths_line(self) -> str:
+        """Return paths."""
+        return " ".join(self._paths) + "\n"
+
+
+class BucketHolder:
+    """Class to hold buckets."""
+
+    def __init__(self, tests_per_bucket: int, bucket_count: int) -> None:
+        """Initialize bucket holder."""
+        self._tests_per_bucket = tests_per_bucket
+        self._bucket_count = bucket_count
+        self._buckets: list[Bucket] = [Bucket() for _ in range(bucket_count)]
+
+    def _flatten(self, tests: TestFolder) -> list[TestFolder | TestFile]:
```

Comment:
This method can be moved to `TestFolder` class and called `all_tests` or something

Reply:
Also the return value is `list[TestFile]`

Reply:
>Also the return value is list[TestFile]

That's incorrect. In the next line the item itself will be added to the list.
See https://github.com/home-assistant/core/pull/114381/files#diff-5d5fd5ade496ffff82ddbbf95f06cf4acebf48abaeb8f774409316cd4d7caf58R46